### ISSUE TITLE
pixel class probability is right?

### ISSUE THREAD

**Issue by scarftdd** [HRNet/HRNet-Semantic-Segmentation#101]
Hello, I have found one problem.
In seg_hr_ocr.py, in class SpatialGather_Module, there two lines below: 
probs = probs.view(batch_size, c, -1)
probs = F.softmax(self.scale * probs, dim=2) # batch, k , hw

probs is [batch,k,hw] = weight = each pixel probabilty, so the probability of all class of one pixel should be 1, so softmax should axis = 1 = k = class, but the dim of softmax  is 2.  I don't know the probability is right.

**Comment by LinJianping** [comment]
I am also a bit confused about this line of code, maybe this is a mistake?

**Comment by shuaiwang812** [comment]
Same question!

**Comment by shuaiwang812** [comment]
It's SPATIAL SOFTMAX. dim=2

**Comment by wzl639** [comment]
> Hello, I have found one problem. In seg_hr_ocr.py, in class SpatialGather_Module, there two lines below: probs = probs.view(batch_size, c, -1) probs = F.softmax(self.scale * probs, dim=2) # batch, k , hw
> 
> probs is [batch,k,hw] = weight = each pixel probabilty, so the probability of all class of one pixel should be 1, so softmax should axis = 1 = k = class, but the dim of softmax is 2. I don't know the probability is right.

holle i have the same question, 

> It's SPATIAL SOFTMAX. dim=2

holle, what is SPATIAL SOFTMAX, i have the same question, i think the dim should be 1,  Can you answer my question， thank you very much

**Comment by wzl639** [comment]
i think i have understand, yes SPATIAL SOFTMAX is used , dim=2

**Comment by amateurj1** [comment]
> i think i have understand, yes SPATIAL SOFTMAX is used , dim=2

Hello!What is SPATIAL SOFTMAX?Could you give more information about it?   F.softmax is SPATIAL SOFTMAX?
